plugins/module_utils/config_utils.py

```diff
@@ -99,3 +104,192 @@ def changed(self) -> bool:
         """
         orig_dict = self._parse_config_from_file()
         return orig_dict != self._config_dict
+
+    def _search_map(self, dictionary, key):
+        """
+        Recursively search for a key in a nested dictionary.
+
+        This function traverses through a nested dictionary structure to find the value associated with the
+        specified key. It searches through all levels of nested dictionaries until it finds the key or
+        exhausts all possibilities.
+
+        Parameters:
+        - dictionary (dict): The dictionary to search through. This can be a multi-level nested dictionary.
+        - key (str): The key to search for in the dictionary.
+
+        Returns:
+        - The value associated with the found key, or None if the key is not found in the dictionary.
+
+        Example:
+        - Given the dictionary `{"interfaces": {"wan": {"if": 'interfaces/wan/if'}}}` and the key `'if'`,
+        the function will return `'interfaces/wan/if'`.
+        """
+
+        if isinstance(dictionary, dict):
+            for k, v in dictionary.items():
+                if k == key:
+                    return v
+                else:
+                    result = self._search_map(v, key)
+                    if result is not None:
+                        return result
+
+    def _get_xpath(self, module: str = None, setting: str = None, map_dict=None):
+        """
+        Retrieve the XPath for a given module and setting from the version-specific mapping.
+
+        This method takes a module and setting as input and looks up their corresponding XPath
+        in a version-specific mapping dictionary. It performs a direct lookup if the setting is
+        top-level within the module or initiates a recursive search if the setting is nested.
+
+        Parameters:
+        - module (str, optional): The name of the module to look up. Defaults to None.
+        - setting (str, optional): The name of the setting within the module to look up. Defaults to None.
+        - map_dict (dict, optional): A pre-fetched version map dictionary. If not provided,
+        it will use the version_map attribute based on the instance's version attribute. Defaults to None.
+
+        Returns:
+        - str or None: The XPath as a string if the module and setting are found within the map. If the
+        module or setting cannot be found, or the inputs are None, it returns None.
+
+        Example:
+        - For an instance with version 'OPNsense 23.1', if version_map['OPNsense 23.1'] contains {'system': {'hostname': 'system/hostname'}},
+        and we call _get_xpath(module='system', setting='hostname'), it will return 'system/hostname'.
+        """
+
+        map_dict = self.version_map.get(self.version)
```

Comment:
This overwrites the `map_dict` param of this method. Maybe the intention was

```
if map_dict is None:
    map_dict = self.version_map.get(self.version)
```

Reply:
I resolved this error.